### ISSUE TITLE
`bookbinder watch` failing with missing subnav root

### ISSUE THREAD

**Issue by RochesterinNYC** [pivotal-cf/bookbinder#108]
Hi, I'm trying to run `bookbinder watch` on the [cloudfoundry/docs-book-cloudfoundry](https://github.com/cloudfoundry/docs-book-cloudfoundry) repo. I do not have all the book sub-components cloned down. 

I'm encountering the following error:

```
± jw |master ✗| → bundle exec bookbinder watch
  copying /Users/pivotal/workspace/docs-layout-repo
Gathering cloudfoundry/docs-bosh
  copying /Users/pivotal/workspace/docs-bosh
Gathering cloudfoundry/docs-buildpacks
  copying /Users/pivotal/workspace/docs-buildpacks
Gathering cloudfoundry/docs-cf-admin
  copying /Users/pivotal/workspace/docs-cf-admin
Gathering cloudfoundry/docs-cf-cli
  copying /Users/pivotal/workspace/docs-cf-cli
Gathering cloudfoundry/docs-cloudfoundry-concepts
  copying /Users/pivotal/workspace/docs-cloudfoundry-concepts
Gathering cloudfoundry/docs-deploying-cf
  skipping (not found) /Users/pivotal/workspace/docs-deploying-cf
Gathering cloudfoundry/docs-dev-guide
  copying /Users/pivotal/workspace/docs-dev-guide
Gathering cloudfoundry/docs-loggregator
  copying /Users/pivotal/workspace/docs-loggregator
Gathering cloudfoundry/docs-running-cf
  copying /Users/pivotal/workspace/docs-running-cf
Gathering cloudfoundry/docs-services
  copying /Users/pivotal/workspace/docs-services
/Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/subnav/json_from_markdown_toc.rb:24:in `get_links': Subnav root not found at: deploying/subnav-index.html (Bookbinder::Subnav::JsonFromMarkdownToc::SubnavRootMissingError)
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/subnav/json_props_creator.rb:11:in `create'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/subnav/subnav_generator.rb:11:in `generate'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/preprocessing/link_to_site_gen_dir.rb:26:in `block in preprocess'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/preprocessing/link_to_site_gen_dir.rb:25:in `each'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/preprocessing/link_to_site_gen_dir.rb:25:in `preprocess'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/preprocessing/preprocessor.rb:17:in `block in preprocess'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/preprocessing/preprocessor.rb:16:in `each'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/preprocessing/preprocessor.rb:16:in `preprocess'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/commands/watch.rb:41:in `run'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/commands/collection.rb:101:in `watch'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/lib/bookbinder/cli.rb:57:in `watch'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/gems/bookbindery-9.4.2/install_bin/bookbinder:5:in `<top (required)>'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/bin/bookbinder:23:in `load'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/bin/bookbinder:23:in `<main>'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/bin/ruby_executable_hooks:15:in `eval'
	from /Users/pivotal/.rvm/gems/ruby-2.3.0/bin/ruby_executable_hooks:15:in `<main>'
```

I can get around the issue by cloning down all the book sub-components.

**Comment by animatedmax** [comment]
Besides docs-book-cloudfoundry and docs-layout-repo, you musty also have an up-to-date copy of the https://github.com/cloudfoundry/docs-deploying-cf repo cloned locally. Bookbinder 9.4.2 can handle missing subnav content, but apparently still needs the subnav root. We've raised this as something to change in the next version of Bookbinder.

**Comment by adasolcan** [comment]
Hi, we are also using bookbinder 9.4.2. We cloned all of the repos that you mentioned and we are seeing the same error but for a different reason: "Subnav root not found at: pivotalcf/installing/pcf-docs.html" 

Is there some workaround to cloning all the repos, when running `bookbinder bind local`? Maybe we can change the subnav_root in the products hash of the config.yml to point to some magic valid subnav_root?

**Comment by animatedmax** [comment]
Hi @adasolcan 
As a workaround, you can provide the subnav generator with files (even blank ones) at the location specified in the config.yml for subnav_root.

For docs-book-pivotalcf, the current config.yml points to docs-pcf-dev/index.html and docs-pcf-gsg/pcf-docs.html as subnav roots. Run the following commands from the docs-book-pivotalcf directory to create these blank files and bookbinder should let you bind locally:

    mkdir ../docs-pcf-dev/ && touch ../docs-pcf-dev/index.html
    mkdir ../docs-pcf-gsg/ && touch ../docs-pcf-gsg/pcf-docs.html

**Comment by animatedmax** [comment]
This issue has been resolved with Bookbinder v9.5.0
